src/listwidget.cpp

```diff
@@ -0,0 +1,113 @@
+#include "listwidget.h"
+
+#include <algorithm>
+
+#include "utils.h"
+
+namespace newsboat {
+
+ListWidget::ListWidget(const std::string& list_name,
+	std::shared_ptr<Stfl::Form> form)
+	: list_name(list_name)
+	, form(form)
+	, num_lines(0)
+{
+}
+
+void ListWidget::set_lines(uint32_t number_of_lines)
+{
+	num_lines = number_of_lines;
+}
+
+bool ListWidget::move_up()
+{
+	uint32_t curpos = get_position();
```

Comment:
I believe this and the rest of the local variables in this file can all be marked `const`.